internal/pkg/consensus/validation.go

```diff
@@ -120,8 +120,8 @@ func (v *DefaultMessageValidator) Validate(ctx context.Context, msg *types.Unsig
 // more value from the actor's balance.
 func canCoverGasLimit(msg *types.UnsignedMessage, actor *actor.Actor) bool {
 	// balance >= (gasprice*gasLimit + value)
-	gascost := specsbig.Mul(abi.NewTokenAmount(msg.GasPrice.AsBigInt().Int64()), abi.NewTokenAmount(int64(msg.GasLimit)))
-	expense := specsbig.Add(gascost, abi.NewTokenAmount(msg.Value.AsBigInt().Int64()))
+	gascost := specsbig.Mul(abi.NewTokenAmount(msg.GasPrice.Int.Int64()), abi.NewTokenAmount(int64(msg.GasLimit)))
```

Comment:
If `msg.GasPrice` is a `big.Int` (or alias) then this should be just `Mul(msg.GasPrice, ...)`

(Note that GasUnits should become an `int64` too, making this yet cleaner).

Reply:
#3773